android/tests_backend/widgets/canvas.py

```diff
@@ -20,6 +20,13 @@ def reference_variant(self, reference):
     def get_image(self):
         return Image.open(BytesIO(self.impl.get_image_data()))
 
+    def test_get_image_data_internal_fail(self, monkeypatch):
+        original_background = self.native.getBackground()
+        self.native.setBackground(None)
+        with pytest.warns(match="Failed to get canvas background"):
+            self.impl.get_image_data()
+        self.native.setBackground(original_background)
```

Comment:
As with the comment on the implementation - *why* is this occurring? Why is the test suite passing with 100% branch coverage prior to this change?